modules/programs/terminator.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.terminator;
+
+  toValue = val:
+    if val == null then
+      "None"
+    else if val == true then
+      "True"
+    else if val == false then
+      "False"
+    else
+      ''"${toString val}"'';
+
+  toConfigObject = let
+    toKey = depth: key:
+      if depth == 0 then key else toKey (depth - 1) "[${key}]";
+    go = depth: obj:
+      flatten (mapAttrsToList (key: val:
+        if isAttrs val then
+          [ (toKey depth key) ] ++ go (depth + 1) val
+        else
+          [ "${key} = ${toValue val}" ]) obj);
+  in obj: concatStringsSep "\n" (go 1 obj);
+
+in {
+  meta.maintainers = [ maintainers.chisui ];
+
+  options.programs.terminator = {
+    enable = mkEnableOption "terminator tiling terminal emulator";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.terminator;
+      defaultText = literalExample "pkgs.terminator";
+      description = "terminator package to install.";
+    };
+
+    config = mkOption {
+      default = null;
+      description = "configuration for terminator";
+      type = types.nullOr (types.submodule {
+        options = let
+          mkConfigOpt = name:
+            (mkOption {
+              default = { };
+              type = types.attrsOf types.anything;
+              description =
+                "configration for ${name}. for options refere to the terminator_config manpage.";
+            });
+        in {
+          global_config = mkConfigOpt "global_config";
+          keybindings = mkConfigOpt "keybindings";
+          profiles = mkConfigOpt "profiles";
+          plugins = mkConfigOpt "plugins";
+        };
```

Comment:
This doesn't make a ton of sense to me, as it seems like this would be handled by using the existing INI format, but I don't quite understand what the config file looks like.

Reply:
Yes, this submodule could be replaces with `attrsOf anything`.